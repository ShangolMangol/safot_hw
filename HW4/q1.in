print "===TEST START===\n";
val test1 = if (runCycle [" *** ", " * * ", " *** "]) = [" * * ","*   *"," * * "] then "PASSED" else "FAILED";
val game = gameOfLife [" *** ", " * * ", " *** "];
game();
game();
game();