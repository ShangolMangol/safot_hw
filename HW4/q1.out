Standard ML of New Jersey (64-bit) v110.99.2 [built: Sun Mar 20 20:10:23 2022]
[opening hw4_q1.sml]
[opening hw4_q1_def.sml]
[opening hw2_q3.sml]
datatype cell = alive | empty
val is_alive = fn :
  cell * cell * cell -> cell * cell * cell -> cell * cell * cell -> cell
val it = () : unit
[opening hw3_q1.sml]
[opening hw3_q1_def.sml]
val toChar = fn : bool -> char
val it = () : unit
val mapState = fn : ('a -> 'b) -> 'a list list -> 'b list list
[autoloading]
[library $SMLNJ-BASIS/basis.cm is stable]
[library $SMLNJ-BASIS/(basis.cm):basis-common.cm is stable]
[autoloading done]
val toString = fn : char list -> string
val frameToState = fn : string list -> bool list list
val stateToFrame = fn : bool list list -> string list
val printFrame = fn : string list -> unit
val it = () : unit
[opening hw3_q2.sml]
[opening hw3_q2_def.sml]
signature KERNEL1D_SIG =
  sig
  type source
  type target
  val kernel : source -> source -> source -> target
  val default : source -> source
end
signature KERNEL2D_SIG =
  sig
  type source
  type target
  val kernel :
      source * source * source
       -> source * source * source -> source * source * source -> target
  val default : source -> source
end
val it = () : unit
functor Kernel1D(Elt: sig
  type source
  type target
  val kernel : source -> source -> source -> target
  val default : source -> source
end) :
sig
  val runKernel : Elt.source list -> Elt.target list
end
val sum = fn : int -> int -> int -> int
structure SumKernel : sig
  val runKernel : Elt.source list -> Elt.target list
end
val it = [3,6,9,7] : int list
exception SizeNotEqual
val zip = fn : 'a list -> 'b list -> 'c list -> ('a * 'b * 'c) list
val it = [(1,4,7),(2,5,8),(3,6,9)] : (int * int * int) list
exception NegativeLength
val fill = fn : 'a -> int -> 'a list
val it = [false,false,false] : bool list
[autoloading]
[autoloading done]
functor Kernel2D(Elt: sig
  type source
  type target
  val kernel :
      source * source * source
       -> source * source * source -> source * source * source -> target
  val default : source -> source
end) :
sig
  val runKernel : source list -> target list list
end
val it = () : unit
val is_alive_bool = fn :
  bool * bool * bool -> bool * bool * bool -> bool * bool * bool -> bool
val run = fn : (unit -> 'a) -> int -> real -> unit
val start_frame =
  ["                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                                                                      #",
   "                     ***  ***  ***  ***   *   ***                     #",
   ...] : string list
val it = () : unit
val runCycle = fn : string list -> string list
val gameOfLife = fn : string list -> unit -> unit
- ===TEST START===
val it = () : unit
val test1 = "PASSED" : string
val game = fn : unit -> unit
 *** 
 * * 
 *** 
val it = () : unit
 * * 
*   *
 * * 
val it = () : unit
     
** **
     
val it = () : unit

