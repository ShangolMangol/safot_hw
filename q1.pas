program PascalTriangle;

type LineArray = array[1..15] of integer;

procedure CreateNextLine(var line: LineArray; var prevLine: LineArray; level: integer);
    var i: integer;
    begin
        i := 2;
        if level <> 1 then
        begin    
            repeat
                line[i] := prevLine[i-1] + prevLine[i];
                i := i + 1;
            until (i > level);
        end;
    end;

var n, i, j: integer;
var line, prevLine: LineArray;

begin
for i := 1 to 15 do
    begin
        line[i] := 1;
        prevLine[i] := 1
    end;
ReadLn(n);
for i := 1 to n do
    begin
        for j := 1 to i-1 do
        begin
            write(line[j], ' ');
        end;
        writeLn(line[i]);
        
        if i <> 15 then
        begin
            for j := 1 to 15 do
            begin
                prevLine[j] := line[j]
            end;
            
            CreateNextLine(line, prevLine, i)

        end;
    end;
end.
