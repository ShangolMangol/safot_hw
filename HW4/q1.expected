val it = () : unit
val test1 = "PASSED" : string
val game = fn : unit -> unit
 *** 
 * * 
 *** 
val it = () : unit
 * * 
*   *
 * * 
val it = () : unit
     
** **
     
val it = () : unit

